portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,25 +14,46 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.util.HashMap;
-// import java.util.ArrayList;
-import com.google.gson.Gson;
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  HashMap<String, String> messages = new HashMap<String, String>();
-  // ArrayList<String> messages = new ArrayList<String>();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final int defaultMaxNumComments = 7;
+  private int maxNumComments = defaultMaxNumComments;
```

Comment:
Note that since you're defining this field at the class level, when a user changes the number, it'll affect all users suing your website. If you only want these comment num changes to affect individual users you should use [session-scoped attributes](https://www.javajee.com/application-request-session-and-page-scopes-in-servlets-and-jsps).

For example here you would call `request.getSession().setAttribute("maxNumComments", maxNumComments);`. And to read it back you call `(int)request.getSession().getAttribute("maxNumComments");`.

Reply:
Oh ok, I'll do that then.